src/modules/package-index.ts

```diff
@@ -12,185 +12,200 @@
-            const packageIndex: PkgIndexEntry[] = JSON.parse(packageIndexFile);
+            const packageIndex: PackageInterface[] = JSON.parse(packageIndexFile);
             return new PackageIndex(packageIndex);
         } catch (error) {
             throw new Error('No package-index.json found.');
         }
     }
 
-    getCores(): PkgIndexEntry[] {
-        return this._packages.filter((pkg: PkgIndexEntry) => pkg.type === 'core');
+    /**
+     * Gets an array of Core instances from the package index.
+     * @returns {Core[]} - Array of Core instances.
+     * @public
+     */
+    getCores(): Core[] {
+        return this._packages
+            .flatMap((pkg: PackageInterface) => pkg.exposedInterfaces)
+            .filter((eif: ExposedInterface): eif is Core => eif.type === 'core');
     }
 
-    getExtensions(): PkgIndexEntry[] {
-        return this._packages.filter((pkg: PkgIndexEntry) => pkg.type === 'extension');
+    /**
+     * Gets an array of Extension instances from the package index.
+     * @returns {Extension[]} - Array of Extension instances.
+     * @public
+     */
+    getExtensions(): Extension[] {
+        return this._packages
+            .flatMap((pkg: PackageInterface) => pkg.exposedInterfaces)
+            .filter((eif: ExposedInterface): eif is Extension => eif.type === 'extension');
     }
 
-    getAvailableLanguages(): string[] {
-        const availableLanguages: string[] = [];
-        this.getCores().forEach((packageEntry: PkgIndexEntry) => {
-            const match = packageEntry.package.match(sdkPattern);
-            if (match) {
-                const language = match[1];
-                availableLanguages.push(language);
-            }
-        });
-        return availableLanguages;
-    }
-
-    getAvailableExamples(): ExampleDescription[] {
-        const availableExamples: ExampleDescription[] = [];
-        this.getCores().forEach((packageEntry: PkgIndexEntry) => {
-            const match = packageEntry.package.match(sdkPattern);
-            if (!match) {
-                return [];
-            }
-            const examples = packageEntry.exposedInterfaces.filter((exposed: ExposedInterface) => exposed.type === 'examples');
-            if (!examples.length) {
-                return [];
-            }
-            for (const example in examples[0].args) {
-                const language = match[1];
-                availableExamples.push({
-                    name: examples[0].args[example].description,
-                    value: examples[0].args[example].id,
-                    language: language,
-                });
-            }
-        });
-        return availableExamples;
+    /**
+     * Gets an array of PackageInterface instances excluding those with specified keywords in their URIs.
+     * @returns {PackageInterface[]} - Array of PackageInterface instances.
+     * @public
+     */
+    getPackages(): PackageInterface[] {
```

Comment:
This check should not be necessary. We want to keep things modular hence we probably need more attributes in the `Extension` interface. Something along the lines of `default: true` to auto-select some parameterless extensions. E.g. CI/CD pipelines, devcontainer-setup and all the runtimes are extensions, but they should be auto-selected.